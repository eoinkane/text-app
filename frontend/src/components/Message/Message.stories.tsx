import React from "react";
import MessageComponent from "./index";
import Container from "@material-ui/core/Container";

export default {
  title: "Message",
  component: MessageComponent,
};

export const Message = () => {
  const text =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ";
  return (
    <Container style={{ backgroundColor: "#f5f5f5", padding: "4vh 4vw" }}>
      <MessageComponent text={text} />
    </Container>
  );
};
