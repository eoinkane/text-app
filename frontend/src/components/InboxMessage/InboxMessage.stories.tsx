import React from "react";
import InboxMessageComponent from "./index";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Message from "../../models/Message";
import User from "../../models/User";

export default {
  title: "InboxMessage",
  component: InboxMessageComponent,
};

export const InboxMessage = () => {
  const currentUser = new User();
  currentUser.init({
    username: "test",
    id: 0,
    firstName: "Jane",
    lastName: "Doe",
  });
  const sentMessage = new Message({
    sender: currentUser,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  const otherParticipant = new User();
  otherParticipant.init({
    username: "test",
    id: 0,
    firstName: "John",
    lastName: "Appleseed",
  });
  const recievedMessage = new Message({
    sender: otherParticipant,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  return (
    <Container style={{ padding: "4vh 4vw" }}>
      <Typography
        style={{
          textAlign: "center",
          paddingBottom: 10,
        }}
      >
        Sent Inbox Message
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          padding: "20px 20px",
          margin: "20px 0",
        }}
      >
        <InboxMessageComponent
          sent
          otherParticipant={otherParticipant}
          message={sentMessage}
        />
      </Container>

      <Typography
        style={{
          textAlign: "center",
          paddingBottom: 10,
        }}
      >
        Recieved Inbox Message
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          padding: "20px 20px",
          margin: "20px 0",
        }}
      >
        <InboxMessageComponent
          recieved
          otherParticipant={otherParticipant}
          message={recievedMessage}
        />
      </Container>
    </Container>
  );
};
