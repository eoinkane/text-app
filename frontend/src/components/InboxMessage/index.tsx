import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { Paper, ListItemText } from "@material-ui/core";
import Message from "../../models/Message";
import User from "../../models/User";

interface InboxMessageBase {
  currentUser: User;
  otherParticipant: User;
  message: Message;
}

interface InboxMessageSent extends InboxMessageBase {
  sent: boolean;
}

interface InboxMessageRecieved extends InboxMessageBase {
  recieved: boolean;
}

type InboxMessageProps = InboxMessageSent | InboxMessageRecieved;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      padding: "2vh 2vw",
      borderRadius: 10,
    },
    messageTypography: {
      overflow: "hidden",
      maxHeight: "3vh",
    },
  })
);

const InboxMessage: React.FC<InboxMessageProps> = ({
  currentUser,
  otherParticipant,
  message,
  ...rest
}) => {
  const classes = useStyles();

  const otherParticipantName = `${
    otherParticipant.get("firstName") as string
  } ${otherParticipant.get("lastName") as string}`;

  let prefix;
  if ((rest as Object).hasOwnProperty("recieved")) {
    if ((rest as InboxMessageRecieved).recieved) {
      const firstName = (message.get("sender") as User).get(
        "firstName"
      ) as string;
      const lastName = (message.get("sender") as User).get(
        "lastName"
      ) as string;
      prefix = `${firstName[0]}${lastName[0]}`;
    }
  } else if ((rest as Object).hasOwnProperty("sent")) {
    if ((rest as InboxMessageSent).sent) {
      prefix = `You`;
    }
  }

  const secondaryText = `${prefix}: ${message.get("message") as string}`;

  return (
    <div className={classes.root}>
      <Paper className={classes.paper} elevation={3}>
        <ListItemText
          primary={otherParticipantName}
          secondary={secondaryText}
          secondaryTypographyProps={{
            className: classes.messageTypography,
          }}
        />
      </Paper>
    </div>
  );
};

export default InboxMessage;
