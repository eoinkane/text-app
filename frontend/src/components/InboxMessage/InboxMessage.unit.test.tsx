import React from "react";
import { shallow, mount } from "enzyme";
import Typography from "@material-ui/core/Typography";

import InboxMessage from "./index";
import Message from "../../models/Message";
import User from "../../models/User";

describe("InboxMessage", () => {
  const currentUser = new User();
  currentUser.init({
    id: 0,
    username: "test",
    firstName: "Jane",
    lastName: "Doe",
  });
  const sentMessage = new Message({
    sender: currentUser,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  const otherParticipant = new User();
  otherParticipant.init({
    id: 0,
    username: "test",
    firstName: "John",
    lastName: "Appleseed",
  });

  const recievedMessage = new Message({
    sender: otherParticipant,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });

  it("Renders ", () => {
    const inboxMessage = shallow(
      <InboxMessage
        sent
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessage).toMatchSnapshot();
  });

  it("Renders with the correct text in sent mode", () => {
    const inboxMessage = mount(
      <InboxMessage
        sent
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );

    expect(inboxMessage).toMatchSnapshot();
    expect(inboxMessage.find(Typography).first().text()).toBe(
      `${otherParticipant.get("firstName") as string} ${
        otherParticipant.get("lastName") as string
      }`
    );
    expect(inboxMessage.find(Typography).last().text()).toBe(
      `You: ${sentMessage.get("message")}`
    );
  });

  it("Renders with the correct text in recieved mode", () => {
    const inboxMessage = mount(
      <InboxMessage
        recieved
        otherParticipant={otherParticipant}
        message={recievedMessage}
      />
    );

    const firstName = otherParticipant.get("firstName") as string;
    const lastName = otherParticipant.get("lastName") as string;
    const initials = `${firstName[0]}${lastName[0]}`;

    expect(inboxMessage).toMatchSnapshot();
    expect(inboxMessage.find(Typography).first().text()).toBe(
      `${otherParticipant.get("firstName") as string} ${
        otherParticipant.get("lastName") as string
      }`
    );
    expect(inboxMessage.find(Typography).last().text()).toBe(
      `${initials}: ${sentMessage.get("message")}`
    );
  });

  it("Applys the correct class for the secondary typography", () => {
    const inboxMessage = mount(
      <InboxMessage
        recieved
        otherParticipant={otherParticipant}
        message={recievedMessage}
      />
    );
    expect(inboxMessage).toMatchSnapshot();

    expect(
      inboxMessage
        .find(Typography)
        .last()
        .hasClass("makeStyles-messageTypography-3")
    ).toBe(true);
  });
});
