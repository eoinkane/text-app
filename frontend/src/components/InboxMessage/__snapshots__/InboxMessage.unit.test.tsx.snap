// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxMessage Applys the correct class for the secondary typography 1`] = `
<InboxMessage
  message={
    Message {
      "data": Object {
        "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
        "sender": User {
          "data": Object {
            "firstName": "John",
            "id": 0,
            "lastName": "Appleseed",
            "username": "test",
          },
        },
      },
    }
  }
  otherParticipant={
    User {
      "data": Object {
        "firstName": "John",
        "id": 0,
        "lastName": "Appleseed",
        "username": "test",
      },
    }
  }
  recieved={true}
>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-2"
      elevation={3}
    >
      <ForwardRef(Paper)
        className="makeStyles-paper-2"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
        elevation={3}
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
        >
          <WithStyles(ForwardRef(ListItemText))
            primary="John Appleseed"
            secondary="JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
            secondaryTypographyProps={
              Object {
                "className": "makeStyles-messageTypography-3",
              }
            }
          >
            <ForwardRef(ListItemText)
              classes={
                Object {
                  "dense": "MuiListItemText-dense",
                  "inset": "MuiListItemText-inset",
                  "multiline": "MuiListItemText-multiline",
                  "primary": "MuiListItemText-primary",
                  "root": "MuiListItemText-root",
                  "secondary": "MuiListItemText-secondary",
                }
              }
              primary="John Appleseed"
              secondary="JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
              secondaryTypographyProps={
                Object {
                  "className": "makeStyles-messageTypography-3",
                }
              }
            >
              <div
                className="MuiListItemText-root MuiListItemText-multiline"
              >
                <WithStyles(ForwardRef(Typography))
                  className="MuiListItemText-primary"
                  component="span"
                  display="block"
                  variant="body1"
                >
                  <ForwardRef(Typography)
                    className="MuiListItemText-primary"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    component="span"
                    display="block"
                    variant="body1"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      John Appleseed
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-messageTypography-3"
                  color="textSecondary"
                  display="block"
                  variant="body2"
                >
                  <ForwardRef(Typography)
                    className="makeStyles-messageTypography-3"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    color="textSecondary"
                    display="block"
                    variant="body2"
                  >
                    <p
                      className="MuiTypography-root makeStyles-messageTypography-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
                    </p>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </ForwardRef(ListItemText)>
          </WithStyles(ForwardRef(ListItemText))>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </div>
</InboxMessage>
`;

exports[`InboxMessage Renders  1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Paper))
    className="makeStyles-paper-2"
    elevation={3}
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="John Appleseed"
      secondary="You: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
      secondaryTypographyProps={
        Object {
          "className": "makeStyles-messageTypography-3",
        }
      }
    />
  </WithStyles(ForwardRef(Paper))>
</div>
`;

exports[`InboxMessage Renders with the correct text in recieved mode 1`] = `
<InboxMessage
  message={
    Message {
      "data": Object {
        "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
        "sender": User {
          "data": Object {
            "firstName": "John",
            "id": 0,
            "lastName": "Appleseed",
            "username": "test",
          },
        },
      },
    }
  }
  otherParticipant={
    User {
      "data": Object {
        "firstName": "John",
        "id": 0,
        "lastName": "Appleseed",
        "username": "test",
      },
    }
  }
  recieved={true}
>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-2"
      elevation={3}
    >
      <ForwardRef(Paper)
        className="makeStyles-paper-2"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
        elevation={3}
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
        >
          <WithStyles(ForwardRef(ListItemText))
            primary="John Appleseed"
            secondary="JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
            secondaryTypographyProps={
              Object {
                "className": "makeStyles-messageTypography-3",
              }
            }
          >
            <ForwardRef(ListItemText)
              classes={
                Object {
                  "dense": "MuiListItemText-dense",
                  "inset": "MuiListItemText-inset",
                  "multiline": "MuiListItemText-multiline",
                  "primary": "MuiListItemText-primary",
                  "root": "MuiListItemText-root",
                  "secondary": "MuiListItemText-secondary",
                }
              }
              primary="John Appleseed"
              secondary="JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
              secondaryTypographyProps={
                Object {
                  "className": "makeStyles-messageTypography-3",
                }
              }
            >
              <div
                className="MuiListItemText-root MuiListItemText-multiline"
              >
                <WithStyles(ForwardRef(Typography))
                  className="MuiListItemText-primary"
                  component="span"
                  display="block"
                  variant="body1"
                >
                  <ForwardRef(Typography)
                    className="MuiListItemText-primary"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    component="span"
                    display="block"
                    variant="body1"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      John Appleseed
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-messageTypography-3"
                  color="textSecondary"
                  display="block"
                  variant="body2"
                >
                  <ForwardRef(Typography)
                    className="makeStyles-messageTypography-3"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    color="textSecondary"
                    display="block"
                    variant="body2"
                  >
                    <p
                      className="MuiTypography-root makeStyles-messageTypography-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      JA: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
                    </p>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </ForwardRef(ListItemText)>
          </WithStyles(ForwardRef(ListItemText))>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </div>
</InboxMessage>
`;

exports[`InboxMessage Renders with the correct text in sent mode 1`] = `
<InboxMessage
  message={
    Message {
      "data": Object {
        "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
        "sender": User {
          "data": Object {
            "firstName": "Jane",
            "id": 0,
            "lastName": "Doe",
            "username": "test",
          },
        },
      },
    }
  }
  otherParticipant={
    User {
      "data": Object {
        "firstName": "John",
        "id": 0,
        "lastName": "Appleseed",
        "username": "test",
      },
    }
  }
  sent={true}
>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-paper-2"
      elevation={3}
    >
      <ForwardRef(Paper)
        className="makeStyles-paper-2"
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
        elevation={3}
      >
        <div
          className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation3 MuiPaper-rounded"
        >
          <WithStyles(ForwardRef(ListItemText))
            primary="John Appleseed"
            secondary="You: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
            secondaryTypographyProps={
              Object {
                "className": "makeStyles-messageTypography-3",
              }
            }
          >
            <ForwardRef(ListItemText)
              classes={
                Object {
                  "dense": "MuiListItemText-dense",
                  "inset": "MuiListItemText-inset",
                  "multiline": "MuiListItemText-multiline",
                  "primary": "MuiListItemText-primary",
                  "root": "MuiListItemText-root",
                  "secondary": "MuiListItemText-secondary",
                }
              }
              primary="John Appleseed"
              secondary="You: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. "
              secondaryTypographyProps={
                Object {
                  "className": "makeStyles-messageTypography-3",
                }
              }
            >
              <div
                className="MuiListItemText-root MuiListItemText-multiline"
              >
                <WithStyles(ForwardRef(Typography))
                  className="MuiListItemText-primary"
                  component="span"
                  display="block"
                  variant="body1"
                >
                  <ForwardRef(Typography)
                    className="MuiListItemText-primary"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    component="span"
                    display="block"
                    variant="body1"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                    >
                      John Appleseed
                    </span>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-messageTypography-3"
                  color="textSecondary"
                  display="block"
                  variant="body2"
                >
                  <ForwardRef(Typography)
                    className="makeStyles-messageTypography-3"
                    classes={
                      Object {
                        "alignCenter": "MuiTypography-alignCenter",
                        "alignJustify": "MuiTypography-alignJustify",
                        "alignLeft": "MuiTypography-alignLeft",
                        "alignRight": "MuiTypography-alignRight",
                        "body1": "MuiTypography-body1",
                        "body2": "MuiTypography-body2",
                        "button": "MuiTypography-button",
                        "caption": "MuiTypography-caption",
                        "colorError": "MuiTypography-colorError",
                        "colorInherit": "MuiTypography-colorInherit",
                        "colorPrimary": "MuiTypography-colorPrimary",
                        "colorSecondary": "MuiTypography-colorSecondary",
                        "colorTextPrimary": "MuiTypography-colorTextPrimary",
                        "colorTextSecondary": "MuiTypography-colorTextSecondary",
                        "displayBlock": "MuiTypography-displayBlock",
                        "displayInline": "MuiTypography-displayInline",
                        "gutterBottom": "MuiTypography-gutterBottom",
                        "h1": "MuiTypography-h1",
                        "h2": "MuiTypography-h2",
                        "h3": "MuiTypography-h3",
                        "h4": "MuiTypography-h4",
                        "h5": "MuiTypography-h5",
                        "h6": "MuiTypography-h6",
                        "noWrap": "MuiTypography-noWrap",
                        "overline": "MuiTypography-overline",
                        "paragraph": "MuiTypography-paragraph",
                        "root": "MuiTypography-root",
                        "srOnly": "MuiTypography-srOnly",
                        "subtitle1": "MuiTypography-subtitle1",
                        "subtitle2": "MuiTypography-subtitle2",
                      }
                    }
                    color="textSecondary"
                    display="block"
                    variant="body2"
                  >
                    <p
                      className="MuiTypography-root makeStyles-messageTypography-3 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      You: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
                    </p>
                  </ForwardRef(Typography)>
                </WithStyles(ForwardRef(Typography))>
              </div>
            </ForwardRef(ListItemText)>
          </WithStyles(ForwardRef(ListItemText))>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </div>
</InboxMessage>
`;
